vertx-mongo-client/src/main/java/io/vertx/ext/mongo/IndexOptions.java

```diff
@@ -0,0 +1,540 @@
+ */
+@DataObject
+public class IndexOptions {
+
+    public static final boolean DEFAULT_BACKGROUD = false;
+    public static final boolean DEFAULT_UNIQUE = false;
+    public static final boolean DEFAULT_SPARSE = false;
+
+    private boolean background;
+    private boolean unique;
+    private String name;
+    private boolean sparse;
+    private Long expireAfterSeconds;
+    private Integer version;
+    private JsonObject weights;
+    private String defaultLanguage;
+    private String languageOverride;
+    private Integer textVersion;
+    private Integer sphereVersion;
+    private Integer bits;
+    private Double min;
+    private Double max;
+    private Double bucketSize;
+    private JsonObject storageEngine;
+    private JsonObject partialFilterExpression;
+
+    /**
+     * Default constructor
+     */
+    public IndexOptions() {
+        background = DEFAULT_BACKGROUD;
+        unique = DEFAULT_UNIQUE;
+        sparse = DEFAULT_SPARSE;
+    }
+
+    /**
+     * Copy constructor
+     *
+     * @param other  the one to copy
+     */
+    public IndexOptions(IndexOptions other) {
+        background = other.background;
+        unique = other.unique;
+        name = other.name;
+        sparse = other.sparse;
+        expireAfterSeconds = other.expireAfterSeconds;
+        version = other.version;
+        weights = other.weights;
+        defaultLanguage = other.defaultLanguage;
+        languageOverride = other.languageOverride;
+        textVersion = other.textVersion;
+        sphereVersion = other.sphereVersion;
+        bits = other.bits;
+        min = other.min;
+        max = other.max;
+        bucketSize = other.bucketSize;
+        storageEngine = other.storageEngine;
+        partialFilterExpression = other.partialFilterExpression;
+    }
+
+    /**
+     * Constructor from JSON
+     *
+     * @param json  the JSON
+     */
+    public IndexOptions(JsonObject json) {
+        background = json.getBoolean("background", DEFAULT_BACKGROUD);
+        unique = json.getBoolean("unique", DEFAULT_UNIQUE);
+        name = json.getString("name");
+        sparse = json.getBoolean("sparse", DEFAULT_SPARSE);
+        expireAfterSeconds = json.getLong("expireAfterSeconds");
+        version = json.getInteger("version");
+        weights = json.getJsonObject("weights");
+        defaultLanguage = json.getString("defaultLanguage");
+        languageOverride = json.getString("languageOverride");
+        textVersion = json.getInteger("textVersion");
+        sphereVersion = json.getInteger("sphereVersion");
+        bits = json.getInteger("bits");
+        min = json.getDouble("min");
+        max = json.getDouble("max");
+        bucketSize = json.getDouble("bucketSize");
+        storageEngine = json.getJsonObject("storageEngine");
+        partialFilterExpression = json.getJsonObject("partialFilterExpression");
+    }
+
+    public com.mongodb.client.model.IndexOptions convertIndexOptions() {
```

Comment:
Convert from `io.vertx.ext.mongo.IndexOptions` to `com.mongodb.client.model.IndexOptions`. It seems this is unnecessary method, I'll move this code to `MongoClientImpl::createIndexWithOptions`